virtual_ecosystem/models/animals/animal_cohorts.py

```diff
@@ -195,145 +188,418 @@ def die_individual(self, number_dead: int, carcass_pool: CarcassPool) -> None:
             LOGGER.warning("No individuals in cohort to forage.")
-            return None  # Early return with no food choice
+            return
+
+        # Herbivore diet
+        if self.functional_group.diet == DietType.HERBIVORE and plant_list:
+            consumed_mass = self.delta_mass_herbivory(
+                plant_list, excrement_pool
+            )  # Directly modifies the plant mass
+            self.eat(consumed_mass)  # Accumulate net mass gain from each plant
+
+        # Carnivore diet
+        elif self.functional_group.diet == DietType.CARNIVORE and animal_list:
+            # Calculate the mass gained from predation
+            consumed_mass = self.delta_mass_predation(
+                animal_list, excrement_pool, carcass_pool
+            )
+            # Update the predator's mass with the total gained mass
+            self.eat(consumed_mass)
 
-        mass_consumed = 0.0  # Initialize to 0.0
+        else:
+            # No appropriate food sources for the diet type
+            raise ValueError(
+                f"No appropriate foods available for {self.functional_group.diet} diet."
+            )
 
-        try:
-            if self.functional_group.diet == DietType.HERBIVORE and plant_list:
-                food_choice = choice(plant_list)
-                mass_consumed = self.eat(food_choice, excrement_pool)
-            elif self.functional_group.diet == DietType.CARNIVORE and animal_list:
-                food_choice = choice(animal_list)
-                mass_consumed = self.eat(food_choice, carcass_pool)
-            else:
-                LOGGER.info("No food available.")
-                food_choice = None  # No food available
+    def theta_i_j(self, animal_list: Sequence[AnimalCohort]) -> float:
+        """Cumulative density method for delta_mass_predation.
 
-        except ValueError as e:
-            raise e
+        The cumulative density of organisms with a mass lying within the same predator
+        specific mass bin as Mi.
 
-        # excrete excess digestive wastes
-        self.excrete(excrement_pool, mass_consumed)
+        Madingley
 
-        return food_choice
+        TODO: current format makes no sense, dig up the details in the supp
+        TODO: update A_cell with real reference
 
-    def eat(self, food: Resource, pool: DecayPool) -> float:
-        """This function handles the mass transfer of a trophic interaction.
+        Args:
+            animal_list: A sequence of animal cohorts that can be consumed by the
+                         predator.
+
+        Returns:
+            The float value of theta.
+        """
+        A_cell = 1.0  # temporary
+        theta = 0.0
 
-        Currently, all this does is call the food's get_eaten method and pass the
-        returned mass value to the consumer.
+        for cohort in animal_list:
+            if self.mass_current == cohort.mass_current:
+                theta += cohort.individuals / A_cell
```

Comment:
As above, this _might_ be faster:
```suggestion
        return sum(
            cohort.individuals / A_cell
            for cohort in animal_list
            if self.mass_current == cohort.mass_current
        )
```